.NET 6: ScriptMethodsGenerator crashes if it encounters a pointer

### Godot version

4.0 b1

### System information

Windows 10 GLES3 3060GTX Mobility

### Issue description

C# code marked unsafe which contains pointers (Notably required for Dear IMGui callbacks) will crash ScriptMethodsGenerator with the following warning in the MSBuild console:

1>CSC : warning CS8785: Generator 'ScriptMethodsGenerator' failed to generate source. It will not contribute to the output and compilation errors may occur as a result. Exception was of type 'NullReferenceException' with message 'Object reference not set to an instance of an object.'

As a result, no C# methods are linked and all mono scripts become silently nonfunctional.



### Steps to reproduce

Demo project attached, or if you want to repro for yourself:

Add AllowUnsafeBlocks to your .csproj properties:

```
<PropertyGroup>
  <TargetFramework>net6.0</TargetFramework> 
  <AllowUnsafeBlocks>True</AllowUnsafeBlocks>
</PropertyGroup>
```
 
Add the following C# script to your project:

```
public partial class DemoMonoCrash : Node
{   
    public unsafe int crashMono(float* data)
    { 
        return 0;
    } 
}

```

Compile and build.

### Minimal reproduction project

[MonoCrash.zip](https://github.com/godotengine/godot/files/9618986/MonoCrash.zip)

--- a/modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators/ExtensionMethods.cs
+++ b/modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators/ExtensionMethods.cs
@@ -29,7 +29,7 @@ namespace Godot.SourceGenerators
         {
             while (symbol != null)
             {
-                if (symbol.ContainingAssembly.Name == assemblyName &&
+                if (symbol.ContainingAssembly?.Name == assemblyName &&
                     symbol.ToString() == typeFullName)
                 {
                     return true;
@@ -47,7 +47,7 @@ namespace Godot.SourceGenerators
 
             while (symbol != null)
             {
-                if (symbol.ContainingAssembly.Name == "GodotSharp")
+                if (symbol.ContainingAssembly?.Name == "GodotSharp")
                     return symbol;
 
                 symbol = symbol.BaseType;

--- a/modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators/MarshalUtils.cs
+++ b/modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators/MarshalUtils.cs
@@ -124,8 +124,8 @@ namespace Godot.SourceGenerators
 
                     if (typeKind == TypeKind.Struct)
                     {
-                        if (type.ContainingAssembly.Name == "GodotSharp" &&
-                            type.ContainingNamespace.Name == "Godot")
+                        if (type.ContainingAssembly?.Name == "GodotSharp" &&
+                            type.ContainingNamespace?.Name == "Godot")
                         {
                             return type switch
                             {
@@ -208,9 +208,9 @@ namespace Godot.SourceGenerators
                         if (type.SimpleDerivesFrom(typeCache.GodotObjectType))
                             return MarshalType.GodotObjectOrDerived;
 
-                        if (type.ContainingAssembly.Name == "GodotSharp")
+                        if (type.ContainingAssembly?.Name == "GodotSharp")
                         {
-                            switch (type.ContainingNamespace.Name)
+                            switch (type.ContainingNamespace?.Name)
                             {
                                 case "Godot":
                                     return type switch
@@ -220,7 +220,7 @@ namespace Godot.SourceGenerators
                                         _ => null
                                     };
                                 case "Collections"
-                                    when type.ContainingNamespace.FullQualifiedName() == "Godot.Collections":
+                                    when type.ContainingNamespace?.FullQualifiedName() == "Godot.Collections":
                                     return type switch
                                     {
                                         { Name: "Dictionary" } =>

